test/packer_basic_example_test.go

```diff
@@ -53,6 +55,61 @@ func TestPackerBasicExample(t *testing.T) {
 	assert.True(t, amiIsPublic)
 }
 
+// An example of how to test the Packer template in examples/packer-basic-example using Terratest
+// with the VarFiles option. This test generates a temporary *.json file containing the value
+// for the `aws_region` variable.
+func TestPackerBasicExampleWithVarFile(t *testing.T) {
+	t.Parallel()
+
+	// Pick a random AWS region to test in. This helps ensure your code works in all regions.
+	awsRegion := terratest_aws.GetRandomStableRegion(t, nil, nil)
+
+	// Create temporary packer variable file to store aws region
+	varFile, err := ioutil.TempFile("", "*.json")
+	assert.NoError(t, err, "Did not expect temp file creation to cause error")
```

Comment:
Can you use `require.NoError` here? The import path is `github.com/stretchr/testify/require`. The difference is that `require` will fail and *exit* the test if it fails the assertion, vs `assert` will *continue* execution (but mark the test as failed).

Reply:
You got it! Thanks for the clarification. That is definitely an important distinction.